internal/datastore/postgres/watch.go

```diff
@@ -139,10 +141,30 @@ func (pgd *pgDatastore) getNewRevisions(
 	return ids, nil
 }
 
-func (pgd *pgDatastore) loadChanges(ctx context.Context, revision xid8) (*datastore.RevisionChanges, error) {
+func (pgd *pgDatastore) loadChanges(ctx context.Context, revisions []postgresRevision) ([]datastore.RevisionChanges, error) {
+	min := revisions[0].tx.Uint
```

Comment:
is there any chance the revisions slice is empty?

Reply:
Not the way the code is currently written. It's literally under a test for that exact condition:

```
			if len(newTxns) > 0 {
				changesToWrite, err := pgd.loadChanges(ctx, newTxns)
```